plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/NessieIcebergClient.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.nessie;
+
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.SchemaTableName;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.EntriesResponse;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.Reference;
+import org.projectnessie.model.Tag;
+
+import javax.annotation.Nullable;
+import javax.inject.Provider;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.plugin.iceberg.IcebergErrorCode.ICEBERG_CATALOG_ERROR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class NessieIcebergClient
+{
+    private final NessieApiV1 nessieApi;
+    private final Provider<UpdateableReference> reference;
+    private final NessieConfig config;
+
+    public NessieIcebergClient(NessieApiV1 nessieApi, NessieConfig config)
+    {
+        this.nessieApi = requireNonNull(nessieApi, "nessieApi is null");
+        this.config = requireNonNull(config, "nessieConfig is null");
+        this.reference = () -> loadReference(config.getDefaultReferenceName(), null);
```

Comment:
Does the client need to know the reference configured for the connector?

Reply:
yes as it needs to operate on the correct branch/tag

Reply:
eventually this is what we might end up having (the catalog providing the branch/tag to the client), but I didn't want to over-complicate the current implementation. That being said, I would prefer to adjust this part of the code once the catalog knows which branch/tag it is being used with then executing a particular SQL